autocert: looking in the wrong (default) directory for renewal

## What happened?

Autocert is looking in the wrong directory to renew. 

## What did you expect to happen?

I expected autocert to renew my existing certificate. 

## How'd it happen?

I have a certificate set to be auto-renewed by autocert. 

## What's your environment like?

- Pomerium version (retrieve with `pomerium --version` or `/ping` endpoint): `{"level":"info","version":"v0.10.0-dirty+d9a224a","time":"2020-08-15T09:42:31-07:00","message":"cmd/pomerium"}`
- Server Operating System/Architecture/Cloud: osx

## What's your config.yaml?

```config.yaml
autocert: true
```

## What did you see in the logs?

```logs
2020/08/15 09:52:31 [NOTICE] Error while checking if certificate for [httpbin.imac.bdd.io] in storage is also expiring soon: open /Users/bdd/.local/share/certmagic/certificates/acme-v02.api.letsencrypt.org-directory/httpbin.imac.bdd.io/httpbin.imac.bdd.io.crt: no such file or directory
2020/08/15 09:52:31 [INFO] [httpbin.imac.bdd.io] Maintenance routine: certificate expires in 464h34m26.1222s; queueing for renewal
2020/08/15 09:52:31 [INFO] [httpbin.imac.bdd.io] Maintenance routine: attempting renewal with 464h34m26.122138s remaining
2020/08/15 09:52:31 [INFO][httpbin.imac.bdd.io] Renew certificate; acquiring lock...
2020/08/15 09:52:31 [INFO][httpbin.imac.bdd.io] Renew: Lock acquired; proceeding...
2020/08/15 09:52:31 [ERROR] attempt 1: open /Users/bdd/.local/share/certmagic/certificates/acme-v02.api.letsencrypt.org-directory/httpbin.imac.bdd.io/httpbin.imac.bdd.io.crt: no such file or directory - retrying in 1m0s (96.016µs/720h0m0s elapsed)...
{"level":"info","service":"identity_manager","time":"2020-08-15T09:52:47-07:00","message":"refreshing directory users"}
2020/08/15 09:53:31 [ERROR] attempt 2: open /Users/bdd/.local/share/certmagic/certificates/acme-v02.api.letsencrypt.org-directory/httpbin.imac.bdd.io/httpbin.imac.bdd.io.crt: no such file or directory - retrying in 2m0s (1m0.001521516s/720h0m0s elapsed)...
2020/08/15 09:55:31 [ERROR] attempt 3: open /Users/bdd/.local/share/certmagic/certificates/acme-v02.api.letsencrypt.org-directory/httpbin.imac.bdd.io/httpbin.imac.bdd.io.crt: no such file or directory - retrying in 2m0s (3m0.002143666s/720h0m0s elapsed)...
2020/08/15 09:57:31 [ERROR] attempt 4: open /Users/bdd/.local/share/certmagic/certificates/acme-v02.api.letsencrypt.org-directory/httpbin.imac.bdd.io/httpbin.imac.bdd.io.crt: no such file or directory - retrying in 5m0s (5m0.003361745s/720h0m0s elapsed)...
```

## Additional context

```
$ file /Users/bdd/.local/share/pomerium/certificates/acme-v02.api.letsencrypt.org-directory/httpbin.imac.bdd.io/httpbin.imac.bdd.io.crt
/Users/bdd/.local/share/pomerium/certificates/acme-v02.api.letsencrypt.org-directory/httpbin.imac.bdd.io/httpbin.imac.bdd.io.crt: PEM certificate
```

```
file /Users/bdd/.local/share/certmagic/certificates/acme-v02.api.letsencrypt.org-directory/httpbin.imac.bdd.io/httpbin.imac.bdd.io.crt
/Users/bdd/.local/share/certmagic/certificates/acme-v02.api.letsencrypt.org-directory/httpbin.imac.bdd.io/httpbin.imac.bdd.io.crt: cannot open `/Users/bdd/.local/share/certmagic/certificates/acme-v02.api.letsencrypt.org-directory/httpbin.imac.bdd.io/httpbin.imac.bdd.io.crt' (No such file or directory)
```

--- a/internal/autocert/manager.go
+++ b/internal/autocert/manager.go
@@ -20,6 +20,7 @@ import (
 var (
 	errObtainCertFailed = errors.New("obtain cert failed")
 	errRenewCertFailed  = errors.New("renew cert failed")
+	defaultStorage      = &certmagic.FileStorage{Path: dataDir()}
 )
 
 // Manager manages TLS certificates.
@@ -37,6 +38,12 @@ type Manager struct {
 
 // New creates a new autocert manager.
 func New(src config.Source) (*Manager, error) {
+	// set certmagic default storage cache, otherwise cert renewal loop will be based off
+	// certmagic's own default location
+	certmagic.Default.Storage = &certmagic.FileStorage{
+		Path: src.GetConfig().Options.AutocertOptions.Folder,
+	}
+
 	mgr := &Manager{
 		src:       src,
 		certmagic: certmagic.NewDefault(),

--- a/internal/autocert/manager.go
+++ b/internal/autocert/manager.go
@@ -20,7 +20,6 @@ import (
 var (
 	errObtainCertFailed = errors.New("obtain cert failed")
 	errRenewCertFailed  = errors.New("renew cert failed")
-	defaultStorage      = &certmagic.FileStorage{Path: dataDir()}
 )
 
 // Manager manages TLS certificates.

